### PR TITLE
add terraform demo

### PR DESCRIPTION
**Pull request by denisj44**
## Please also consider adding a line to [CHANGELOG.md](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/CHANGELOG.md)

Version of at least Jenkins 2.73 is needed to run Job DSL plugin that's why the pom.xml would need to be updated.

### CODE REVIEW

**Comment by denisj44** [comment]
@reviewbybees

File: pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
   <properties>
     <revision>1.3</revision>
     <changelist>-SNAPSHOT</changelist>
-    <jenkins.version>2.60.3</jenkins.version>
+    <jenkins.version>2.73.3</jenkins.version>
```
Comment (casz) [inline pom.xml]: Yikes! Without changelog or pull request description saything why this change is needed I cannot approve.

Comment (denisj44) [reply pom.xml]: I added the explanation in the PR description. I hope it helps.

**Comment by casz** [comment]
2.73 is not a LTS. 😭

**Comment by casz** [comment]
2.73 is not a LTS. 😭

File: pom.xml (hunk 1 above)
Comment (ndeloof) [reply pom.xml]: change seems unrelated to the proposed contribution, please void mixing content.
also, I don't get the point: we already run integration test of jobdsl plugin without this core version bump being needed.

**Comment by denisj44** [comment]
Change made.
Thank you.

**Comment by ndeloof** [comment]
LGTM 💋
Please just rebase (and maybe squash) so I can merge
